src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -518,6 +518,26 @@ public function resolveConcatType(Type $left, Type $right): Type
 		return new StringType();
 	}
 
+	protected function appendAccessoryCasedStringTypes($leftStringType, $rightStringType, array &$accessoryTypes): void
+	{
+		$lower = $leftStringType->isLowercaseString()->and($rightStringType->isLowercaseString())->yes();
+		$upper = $leftStringType->isUppercaseString()->and($rightStringType->isUppercaseString())->yes();
+
+		if ($lower && $upper) {
```

Comment:
To me, this if should be removed.
You can both add lowercase and uppercase.

Reply:
Removed in b6b7b49d9